util/ranger/detacher.go

```diff
@@ -201,23 +255,91 @@ func detachCNFCondAndBuildRangeForIndex(sctx sessionctx.Context, conditions []ex
 		shouldReserve: lengths[eqOrInCount] != types.UnspecifiedLength,
 	}
 	if considerDNF {
+		pointRes, offset, err := extractIndexPointRangesForCNF(sctx, conditions, cols, lengths)
+		if err != nil {
+			return nil, err
+		}
+		if pointRes != nil {
+			if len(pointRes.Ranges) == 0 {
+				return &DetachRangeResult{}, nil
```

Comment:
Is there a situation where &DetachRangeResult{} is returned to the caller and causes an error? Because usually the caller will only check whether it is nil.

Reply:
Conventionally, when we return an empty `DetachRangeResult`, it means that the filter is constantly false, for example, https://github.com/pingcap/tidb/blob/master/util/ranger/detacher.go#L187, so if we always follow this convention, IMHO there should be no problem?

Reply:
For example, https://github.com/pingcap/tidb/blob/5153a4155a9ac1ad6c7266e00239819e8dc1ba13/planner/core/rule_max_min_eliminate.go#L81 may receive empty `DetachRangeResult` by https://github.com/pingcap/tidb/blob/5153a4155a9ac1ad6c7266e00239819e8dc1ba13/util/ranger/detacher.go#L371, but it doesn't check and use the return value directly.

Reply:
Hmmm, the example you give has no problem since `result.EqCondCount` is 0? I get your concerns, it is true that returning an empty `DetachRangerResult` is not that graceful, we can refactor this later in another PR, like say introducing a new field into `DetachRangerResult` to indicate those constantly false filters. Since there already exists codes returning empty `DetachRangerResult`, and the refactor is not related to the problem this PR aims to solve, I prefer not to do it in this PR.

Reply:
OK. This sounds great. I have no other questions.